server.go

```diff
@@ -0,0 +1,565 @@
+package sftp
+
+// sftp server counterpart
+
+import (
+	"encoding"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"sync"
+	"syscall"
+	"time"
+)
+
+const (
+	sftpServerWorkerCount = 8
+)
+
+// Server is an SSH File Transfer Protocol (sftp) server.
+// This is intended to provide the sftp subsystem to an ssh server daemon.
+// This implementation currently supports most of sftp server protocol version 3,
+// as specified at http://tools.ietf.org/html/draft-ietf-secsh-filexfer-02
+type Server struct {
+	in            io.Reader
+	out           io.WriteCloser
+	outMutex      *sync.Mutex
+	debugStream   io.Writer
+	debugLevel    int
+	readOnly      bool
+	rootDir       string
+	lastId        uint32
+	pktChan       chan rxPacket
+	openFiles     map[string]serverOpenFile
+	openFilesLock *sync.RWMutex
+	handleCount   int
+	maxTxPacket   uint32
+	workerCount   int
+}
+
+type serverOpenFile struct {
+	*os.File
+	path string
+}
+
+func (svr *Server) nextHandle(path string, f *os.File) string {
+	svr.openFilesLock.Lock()
+	defer svr.openFilesLock.Unlock()
+	svr.handleCount++
+	handle := fmt.Sprintf("%d", svr.handleCount)
```

Comment:
Why stringify this, can you make svr.OpenFiles a map of int to *serverOpenFile ?